src/core/infer.ml

```diff
@@ -132,49 +181,51 @@ let rec infer : ctxt -> term -> term = fun ctx t ->
+      ----------------------------
+         ctx ⊢ Meta(m,e) ⇒ a      *)
   | Meta(m,ts)   ->
-      infer ctx (term_of_meta m ts)
+      (* The type of [Meta(m,ts)] is the same as the one obtained by applying
+         to [ts] a new symbol having the same type as [m]. *)
+      let s = Sign.new_sym (meta_name m) !(m.meta_type) in
+      infer ctx (Array.fold_left (fun acc t -> Appl(acc,t)) (Symb s) ts)
 
-(** [check ctx t c] checks that the term [t] has type [c] in context
-   [ctx], possibly under some constraints recorded in [constraints]
-   using [conv]. [ctx] must be well-formed and [c] well-sorted. This
-   function never fails (but constraints may be unsatisfiable). *)
-and check : ctxt -> term -> term -> unit = fun ctx t c ->
-  if !log_enabled then log_infr "check %a : %a" pp t pp c;
-  conv ctx (infer ctx t) c
-
-(** [infer ctx t] returns a pair [(a,cs)] where [a] is a type for the
-   term [t] in the context [ctx], under unification constraints [cs].
-   In other words, the constraints of [cs] must be satisfied for [t]
-   to have type [a]. [ctx] must be well-formed. This function never
+(** [check ctx t a] checks that the term [t] has type [a] in context [ctx],
+   possibly under some constraints recorded in [constraints] using
+   [conv]. [ctx] must be well-formed and [a] well-sorted. This function never
    fails (but constraints may be unsatisfiable). *)
-let infer : ctxt -> term -> term * unif_constrs = fun ctx t ->
+and check : ctxt -> term -> term -> unit = fun ctx t a ->
+  if !log_enabled then log_infr "check %a : %a" pp_term t pp_term a;
+  conv ctx (infer ctx t) a
+
+(** [infer ctx t] returns a pair [(a,cs)] where [a] is a type for the term [t]
+   in the context [ctx] under unification constraints [cs].  In other words,
+   the constraints of [cs] must be satisfied for [t] to have type [a]. [ctx]
+   must be well-formed. This function never fails (but constraints may be
+   unsatisfiable). *)
+let infer : ctxt -> term -> term * constr list = fun ctx t ->
   Stdlib.(constraints := []);
   let a = infer ctx t in
   let constrs = Stdlib.(!constraints) in
   if !log_enabled then
     begin
-      log_infr (gre "infer %a : %a") pp t pp a;
+      log_infr (gre "infer %a : %a") pp_term t pp_term a;
       List.iter (log_infr "  if %a" pp_constr) constrs;
     end;
   Stdlib.(constraints := []);
   (a, constrs)
 
-(** [check ctx t c] checks returns a list [cs] of unification
-   constraints for [t] to be of type [c] in the context [ctx]. The
-   context [ctx] must be well-typed, and the type [c]
-   well-sorted. This function never fails (but constraints may be
-   unsatisfiable). *)
-let check : ctxt -> term -> term -> unif_constrs = fun ctx t c ->
+(** [check ctx t c] checks returns a list [cs] of unification constraints for
+   [t] to be of type [c] in the context [ctx]. The context [ctx] must be
+   well-typed, and the type [c] well-sorted. This function never fails (but
+   constraints may be unsatisfiable). *)
```

Comment:
Comment not indented correctly.

Reply:
We don't have the same environment and editor. Next time, it's better that you fix the indentation yourself.